### PR TITLE
made .enchant easier to use

### PR DESCRIPTION
**Pull request by hYdos**
i made it so .enchant performs all the steps for nopacket so people dont keep asking questions on HoW DoEs ThIs WoRk

### CODE REVIEW

File: Memory/Hooks.cpp [hunk 1]
```diff
@@ -574,6 +575,12 @@ signed int Hooks::AppPlatform_getGameEdition(__int64 _this)
 	return oGetEditon(_this);
 }
 
+//code wip
+void Hooks::ToggleAutoNoPacket()
+{
+	autoNopacket = !autoNopacket;
+}
+
```
Comment (SurvirvorHRG) [inline Memory/Hooks.cpp]: ghetto code

File: Horion/Command/Commands/EnchantCommand.cpp [hunk 2]
```diff
@@ -136,14 +137,16 @@ bool EnchantCommand::execute(std::vector<std::string>* args)
...
 		}
 		else
-			clientMessageF("%sEnchant failed, try using a lower enchant-level", RED);
+			Hooks::ToggleAutoNoPacket();
+			//clientMessageF("%sEnchant failed, try using a lower enchant-level", RED);
+			//above line commented out because can occur when enchant is successful
 
 		free(EnchantData);
 	}
+	Hooks::ToggleAutoNoPacket();
```
Comment (SurvirvorHRG) [inline Horion/Command/Commands/EnchantCommand.cpp]: use moduleMgr->getModule<NoPacket>()->setEnabled(true) instead of that

Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp]: Sure as soon as I get on my laptop I’ll add that

**Comment by hYdos** [comment]
any more requests before i run out of data?

File: Memory/Hooks.h [hunk 3]
```diff
@@ -36,7 +36,7 @@ class Hooks {
 private:
 	bool shouldRender = true;
 public:
-	
+	static void ToggleAutoNoPacket();
```
Comment (ghost) [inline Memory/Hooks.h]: remove pls

File: Horion/Command/Commands/EnchantCommand.cpp [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #include "EnchantCommand.h"
-
+#include "../../../Memory/Hooks.h"
```
Comment (ghost) [inline Horion/Command/Commands/EnchantCommand.cpp]: remove pls

File: Horion/Command/Commands/EnchantCommand.cpp [hunk 5]
```diff
@@ -93,6 +93,7 @@ bool EnchantCommand::execute(std::vector<std::string>* args)
 	}
 	if (strcmp(args->at(1).c_str(), "all") == 0)
 	{
+		moduleMgr->getModuleByName("noPacket")->setEnabled(true);
```
Comment (ghost) [inline Horion/Command/Commands/EnchantCommand.cpp]: Use getModule<>()

File: Horion/Command/Commands/EnchantCommand.cpp [hunk 6]
```diff
@@ -112,14 +113,15 @@ bool EnchantCommand::execute(std::vector<std::string>* args)
...
-				g_Data.getLocalPlayer()->sendInventory();
+				g_Data.getLocalPlayer()->setOffhandSlot(item);
 			}
 			free(EnchantData);
 		}
 		clientMessageF("%sEnchant successful!", GREEN);
 	}
 	else
 	{
+		moduleMgr->getModuleByName("noPacket")->setEnabled(true);
```
Comment (ghost) [inline Horion/Command/Commands/EnchantCommand.cpp]: Use getModule<>()

**Review by ghost — CHANGES_REQUESTED** [review]

File: Horion/Command/Commands/EnchantCommand.cpp (hunk 5 above)
Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp]: ok il do those In 7 mins max

File: Horion/Command/Commands/EnchantCommand.cpp (hunk 4 above)
Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp]: done

File: Memory/Hooks.h (hunk 3 above)
Comment (hYdos) [reply Memory/Hooks.h]: also done

**Comment by hYdos** [comment]
hmm it still says changes requested after they are all resolved. weird

File: Horion/Command/Commands/EnchantCommand.cpp (hunk 6 above)
Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp]: done

File: Horion/Command/Commands/EnchantCommand.cpp (hunk 5 above)
Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp]: done

File: Memory/Hooks.h (hunk 3 above)
Comment (hYdos) [reply Memory/Hooks.h]: done

**Comment by hYdos** [comment]
added auto .give for people who cant follow steps and just to make it faster

File: Horion/Command/Commands/GiveCommand.cpp [hunk 7]
```diff
@@ -112,7 +119,8 @@ bool GiveCommand::execute(std::vector<std::string>* args)
 		yot->count = count;
 		
 
-	inv->addItemToFirstEmptySlot(yot);
+	g_Data.getLocalPlayer()->setOffhandSlot(yot);
```
Comment (SurvirvorHRG) [inline Horion/Command/Commands/GiveCommand.cpp]: lol better to delete .give

Comment (hYdos) [reply Horion/Command/Commands/GiveCommand.cpp]: wdym like use .setoffhand?

File: Horion/Command/Commands/EnchantCommand.cpp [hunk 8]
```diff
@@ -136,14 +137,16 @@ bool EnchantCommand::execute(std::vector<std::string>* args)
...
 			clientMessageF("%sEnchant successful!", GREEN);
 		}
 		else
-			clientMessageF("%sEnchant failed, try using a lower enchant-level", RED);
+			//clientMessageF("%sEnchant failed, try using a lower enchant-level", RED);
+			//above line commented out because can occur when enchant is successful
 
 		free(EnchantData);
 	}
+	moduleMgr->getModule<NoPacket>()->setEnabled(false);
```
Comment (SurvirvorHRG) [inline Horion/Command/Commands/EnchantCommand.cpp:149]: u don't need to enable nopacket just use setoffhand i think it will work

Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp:149]: ok ill try that

Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp:149]: should i delete setoffhand?

Comment (SurvirvorHRG) [reply Horion/Command/Commands/EnchantCommand.cpp:149]: no

Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp:149]: ok

File: Horion/Command/Commands/GiveCommand.cpp (hunk 7 above)
Comment (SurvirvorHRG) [reply Horion/Command/Commands/GiveCommand.cpp]: now .give and .soh are the same things so it doens't make sense to have two commands that does the same things

Comment (hYdos) [reply Horion/Command/Commands/GiveCommand.cpp]: removed .give because .setoffhand works better

Comment (hYdos) [reply Horion/Command/Commands/GiveCommand.cpp]: should i make .give alias of .setoffhand?

**Comment by hYdos** [comment]
any more requests?

**Comment by hYdos** [comment]
wasnt getting pulled and got other ideas

**Pull request closed without merging**